### PR TITLE
CAMS-122 Automate clean up of Azure resources

### PR DESCRIPTION
**Pull request by magile32**
# Purpose

Automate the clean up of Azure resources in the Flexion environment

# Major Changes

- Create new GHA workflow to execute clean up script
- Update deployment scripts to include tags on Azure resource group used for application resources. Information included is branch name and hash id used to create resource groups.
- Add event trigger for new GHA clean up workflow on pull request close activities

# Testing/Validation

- Verify Azure resource tagging with expected values
- Verify new GHA workflow remove Azure resources by id

# Notes

This workflow is only applicable to Flexion environment to clean up development branches that was deployed into Azure.

### CODE REVIEW

File: .github/workflows/azure-remove-branch.yml [hunk 1]
```diff
@@ -1,15 +1,71 @@
...
+          branchHashId=$(echo -n ${{ github.event.pull_request.head.ref }} | openssl sha256 | awk '{print $2}')
+          echo "hashId=$branchHashId" >> $GITHUB_OUTPUT
+
+      - name: Validate resources
+        id: check
+        run: |
+          target=${{ inputs.hashId == '' && steps.hash.outputs.hashId || inputs.hashId  }}
+          echo "targetBranchHashId=$target" >> $GITHUB_OUTPUT
+
+          echo "Does resources with $target exist?"
```
Comment (btposey) [inline .github/workflows/azure-remove-branch.yml:51]: nitpick: "Does" => "Do"

File: .github/workflows/azure-remove-branch.yml [hunk 2]
```diff
@@ -1,15 +1,71 @@
...
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+          environment: ${{ vars.AZURE_ENVIRONMENT }}
+      - name: List existing branch deployments
+        run: |
+          echo "List Azure resource groups of current branch deployment"
+          az group list --query "[?tags.branchName].{ Name:name Branch:tags.branchName HashId:tags.branchHashId }" -o table
+
+  check:
+    runs-on: ubuntu-latest
+    if: inputs.hashId != '' || github.event.action == 'closed'
```
Comment (jamesobrooks) [inline .github/workflows/azure-remove-branch.yml:28]:
```
on:
  pull_request:
    types: [ closed ]
```
With that, isn't the second part of this condition always true?

File: .github/workflows/azure-remove-branch.yml (hunk 1 above)
Comment (magile32) [reply .github/workflows/azure-remove-branch.yml]: Updated

File: ops/cloud-deployment/lib/subscription/ustp-cams-rg.bicep [hunk 3]
```diff
@@ -29,5 +36,11 @@ module resourceGroup './resource-group-deploy.bicep' = [for item in resourceGrou
   params: {
     location: location
     resourceGroupName: item.name
+    tags: isBranchDeployment ? {
+      // Expected tags to set on Azure Resource Group for resources deployed for branch deploys
```
Comment (jamesobrooks) [inline ops/cloud-deployment/lib/subscription/ustp-cams-rg.bicep:40]: This comment seems unnecessary.

**Review by jamesobrooks — APPROVED** [review]

**Review by fmaddenflx — COMMENTED** [review]
I didn't see any issues to concern me.  Thumbs up.

**Review by amorrow-flexion — APPROVED** [review]

**Review by fmaddenflx — APPROVED** [review]

File: .github/workflows/azure-remove-branch.yml (hunk 2 above)
Comment (magile32) [reply .github/workflows/azure-remove-branch.yml:28]: Had discussion and this is okay for now. Aware that there might be multiple pull requests for same branch that may get merged in and will remove a deployment branch that may still have work in progress. Will design for this later on if it becomes a problem.

**Pull request merged**